.github/workflows/workflow.yaml

```diff
@@ -91,7 +91,7 @@ jobs:
     name: Build and publish Python distributions 📦 to PyPI and TestPyPI
     runs-on: ubuntu-latest
     needs: ["test", "docs"]
-    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' || github.ref == 'refs/heads/devel'}}
```

Comment:
the path to master and devel here make sure that only on push to these branches run the build steps. Do they address the issue?

Reply:
While exploring the separate workflow file that only runs on push to master or devel, especially for build steps I also came across another way to achieve the same thing of only building and publishing when pushed to master/devel using this site as reference: https://kevsoft.net/2020/06/10/running-github-action-steps-and-jobs-only-on-push-to-master.html (Especially section called Combining expressions)
Let me know what you think about the changes done. Above is the place where the changes are done.